pyproject.toml

```diff
@@ -138,6 +138,51 @@ good-names = [
 output-format = "colorized"
 score = "no"
 
+[tool.ruff]
+target-version = "py311"
+
+[tool.ruff.lint]
+select = [
+  "C", "E", "F", "I", "R", "W", "D"
+]
+ignore = [
+  # Missing docstrings.
+  "D100","D101","D102","D103","D104","D105","D106","D107",
+
+  # "No blank lines allowed after function docstring" conflicts with the
+  # Black code formatter which insists on inserting blank lines after
+  # function docstrings.
+  "D202",
+
+  # "1 blank line required before class docstring" conflicts with another
+  # pydocstyle rule D211 "No blank lines allowed before class docstring".
+  "D203",
+
+  # "Multi-line docstring summary should start at the first line"
+  # and "Multi-line docstring summary should start at the second line".
+  # These two rules conflict with each other so you have to disable one of them.
+  # How about we disable them both? PEP 257 says either approach is okay:
+  #
+  # > The summary line may be on the same line as the opening quotes or on
+  # > the next line.
+  # >
+  # > https://peps.python.org/pep-0257/#multi-line-docstrings
+  "D212",
+  "D213",
+
+  "E501", # line-too-long
+
+  "F401",  # unused import
```

Comment:
I'd like to avoid blanket-disabling unused import. It's one of the most basic and useful lints. Also we didn't do that in the pylint configuration. I see we do intentionally have unused imports in various `__init__.py` files, because they are in fact re-exports, which pylint accepted. How did that work?

Reply:
Related upstream discussion - https://github.com/astral-sh/ruff/issues/10253, which discusses workarounds.

Reply:
Adding a `__all__` to `__init__.py` files, or the re-export convention mentioned in the above issue, would be slightly annoying boilerplate, but I'd rather do that than blanket-disable this rule for the whole project.

Reply:
```
[tool.ruff.per-file-ignores]
"__init__.py" = ["F401"]
```
will help with the transition

Reply:
Good spot about the unused-import, I agree that's a useful one

Reply:
A `per-file-ignore` rule works for me.